test/src/core/mimedata.spec.ts

```diff
@@ -0,0 +1,113 @@
+/*-----------------------------------------------------------------------------
+| Copyright (c) 2014-2016, PhosphorJS Contributors
+|
+| Distributed under the terms of the BSD 3-Clause License.
+|
+| The full license is in the file LICENSE, distributed with this software.
+|----------------------------------------------------------------------------*/
+import expect = require('expect.js');
+
+import {
+  MimeData
+} from '../../../lib/core/mimedata';
+
+
+describe('core/mimedata', () => {
+
+  describe('MimeData', () => {
+
+    describe('#types()', () => {
+
+      it('should get an array of the MIME types contained within the dataset', () => {
+        let data = new MimeData();
+        data.setData('foo', 1);
+        data.setData('bar', '1');
+        data.setData('baz', { foo: 1, bar: 2 });
+        expect(data.types()).to.eql(['foo', 'bar', 'baz']);
+      });
+
```

Comment:
need a test which guarantees `types()` returns keys in order of calling `setData` (most recent last). 

e.g.
```
setData('foo', 1)
setData('bar', 2)
setData('baz', 3)
setData('bar', 4)
clearData('foo')
setData('foo', 9)
types() === ['baz', 'bar', 'foo']
```